### ISSUE TITLE
HACS support

### ISSUE THREAD

**Issue by jkkataja** [martinarva/dynamic_energy_cost#6]
Appreciate your worka and he fact that this your first this kind of project.

I think it would be great if you would add HACS support for your work.

cheers

**Comment by martinarva** [comment]
Thanks. I will take some time and dive into HACS requirements and what I have to do next week.

**Comment by whi-tw** [comment]
HACS is working perfectly for me with a custom repository FYI:


<img width="278" alt="image" src="https://github.com/martinarva/dynamic_energy_cost/assets/3595020/ff39e39e-d87b-4e59-a85d-a78e5977a0e5">

![image](https://github.com/martinarva/dynamic_energy_cost/assets/3595020/2a829d21-b466-464d-b5b9-c0f58bff5dbf)

**Comment by martinarva** [comment]
> HACS is working perfectly for me with a custom repository FYI:
> 
> <img alt="image" width="278" src="https://private-user-images.githubusercontent.com/3595020/328963866-ff39e39e-d87b-4e59-a85d-a78e5977a0e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDcwMjMsIm5iZiI6MTcxNTI0NjcyMywicGF0aCI6Ii8zNTk1MDIwLzMyODk2Mzg2Ni1mZjM5ZTM5ZS1kODdiLTRlNTktYTg1ZC1hNzhlNTk3N2EwZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMDkyNTIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjZjNWY1NzEwZTcwMDkxMjNkYjBjZDg1MDViNzY1MGI2ZmJlZjMzNjI2MWQ2YWUyZWNlYzc5NTg4ZjlmMzQ3ZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ysO4EiXRLja53Yzh18N2M4S_-WgbPh1VYr-QO9QgsKA">
> ![image](https://private-user-images.githubusercontent.com/3595020/328963241-2a829d21-b466-464d-b5b9-c0f58bff5dbf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUyNDcwMjMsIm5iZiI6MTcxNTI0NjcyMywicGF0aCI6Ii8zNTk1MDIwLzMyODk2MzI0MS0yYTgyOWQyMS1iNDY2LTQ2NGQtYjViOS1jMGY1OGJmZjVkYmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDlUMDkyNTIzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjhjZDhjMGFiY2Q4YzlhZjEzYzliNzY5YjA4NjFkODgyYmE0OGJjODU4Yjg1OWNiMmY2OTdiOTkzOWM0OTNjOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EdyoFXNxWXdcPyPJpQahNteTOyih1b9Vj6MqBrPJ4MI)

Oh, great. Did not think about that custom repo will to the trick until i put it officially to HACS

**Comment by jkkataja** [comment]
Yeah it looks to be working, information is missing, that could be added at some stage?

**Comment by dunxd** [comment]
Thought this could do with an icon. Here is something I knocked up in draw.io.
PNG Version ![Dynamic Energy Cost Icon](https://github.com/martinarva/dynamic_energy_cost/assets/16608541/af750f9d-5c77-4c07-aa31-eafdb9fc9d64)
SVG Version ![Dynamic Energy Cost Icon](https://github.com/martinarva/dynamic_energy_cost/assets/16608541/a918547c-d728-4a96-810c-6043ee069cf5)

**Comment by BDVGitHub** [comment]
Here is how to put it officially to HACS https://www.hacs.xyz/docs/publish/include/
